modules/core/plan/src/Form/PlanRecordForm.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Drupal\plan\Form;
+
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Form controller for plan_record entities.
+ *
+ * @ingroup plan
+ */
+class PlanRecordForm extends ContentEntityForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildForm($form, $form_state);
+
+    // Hide the plan reference field.
+    $form['plan']['#access'] = FALSE;
```

Comment:
This works for now but does feel a little heavy because I think this will always remove the `plan` field even if an entity form display is configured to use it. Probably OK for now, but I imagine there could be valid use-cases to expose this plan field, especially when creating new plan records. I guess that could be one potential improvement - only remove the field if `!$this->entity->isNew()`?

Otherwise, I'd actually be interested in seeing if we could leverage bundle classes to let some of this logic be determined on a per-bundle case. That way if crop plans never want the `plan` field exposed because there is a separate UI for that, the `crop_planting` bundle of `plan_record` could enforce that. I might experiment with this idea but mostly just mentioning it here, it doesn't need to block this PR.

Reply:
It's true this is heavy, and maybe it shouldn't even be part of this initial commit. My original motivation was so that I could display the `plan_record` edit form in an off-canvas dialog for easy editing in the crop plan module. We can achieve the same thing in downstream code where necessary though (eg: via `hook_form_alter()`) so maybe we shouldn't have any opinion about it in core to start.

Even an opinion involving `!$this->entity->isNew()` might not be valid in some cases. So perhaps we just start simple and take a light touch.

Reply:
Fixup commit: https://github.com/farmOS/farmOS/pull/781/commits/5a303513735d30f2bc360736ef35d41f2987af01